k3s/k3s-init.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+# Check k3s installation
+k3s_path=$(which k3s)
+if [ -x "$k3s_path" ]; then
+    echo "k3s is already installed"
+    exit 0
+fi
+
+# Install k3s
+$ curl -sfL https://get.k3s.io | K3S_KUBECONFIG_MODE="644" sh -s -
+
+K3S_KUBECONFIG=/etc/rancher/k3s/k3s.yaml
+
+# Check if kubectl is installed
+echo "Checking if kubectl is in PATH..."
+kubectl_path=$(which kubectl)
+if [ -x "$kubectl_path" ]; then
+    echo "kubectl is installed, merging kubeconfig files..."
+    
+    # Backup the existing kubeconfig file if it exists
+    if [ -f ~/.kube/config ]; then
+        KUBECONFIG_BACKUP=~/.kube/config.backup
+        cp ~/.kube/config $KUBECONFIG_BACKUP
+        echo "Existing kubeconfig backed up to $KUBECONFIG_BACKUP"
+    fi
+
+    # Merge kubeconfig files
+    env KUBECONFIG=~/.kube/config:$K3S_KUBECONFIG kubectl config view --merge --flatten > /tmp/config
+    mv /tmp/config ~/.kube/config
```

Comment:
Minor thing.

Note that this mv under the hood will delete the existing ~/.kube/config and then move /tmp/config into it's place... But that won't preserve the file permissions and attrs of the original file, e.g. the group ownership could change if the file created under /tmp was created with a different group than the one under ~/.kube.

An alternate approach to avoid this potential loss of state is to overwrite the existing file with the new content:
```suggestion
    cat /tmp/config > ~/.kube/config && rm /tmp/config
```

The use of `&& rm /tmp/config` also ensures that we delete the new config file if we successfully overwrite, but leaves it there to debug if something went wrong.
However if there is a risk of sensitive data leaking by leaving the file in /tmp we should use `; rm -f /tmp/config` instead.

Reply:
An alternative approach that allows you to keep using the `mv` approach is to pre-create the tmp file using a `cp -p <orig_file> <tmp_file>` before overwriting it with the kubectl merge... That way the tmp file is created with the same perms and file attrs as the original, and the mv <tmp_file> <orig_file> doesn't lose any information.